hash.c

```diff
@@ -4830,18 +4857,32 @@ static st_table *rb_init_identtable_with_size(st_index_t size);
 
 /*
  *  call-seq:
- *     hsh.compare_by_identity -> hsh
+ *    hash.compare_by_identity -> self
  *
- *  Makes <i>hsh</i> compare its keys by their identity, i.e. it
- *  will consider exact same objects as same keys.
+ *  Sets +self+ to consider only identity in comparing keys;
+ *  two keys are considered the same only if they are the same object;
+ *  returns +self+.
  *
- *     h1 = { "a" => 100, "b" => 200, :c => "c" }
- *     h1["a"]        #=> 100
- *     h1.compare_by_identity
- *     h1.compare_by_identity? #=> true
- *     h1["a".dup]    #=> nil  # different objects.
- *     h1[:c]         #=> "c"  # same symbols are all same.
+ *  By default, these two object are considered to be the same key,
```

Comment:
How about:

> By default, these two objects are considered to be the same key,
> So +s1+ will overwrite +s0+.

to fix "object" pluralization and be more explicit about the example below?